### ISSUE TITLE
MQTT Source does not listen on defined topics

### ISSUE THREAD

**Issue by sobychacko** [spring-cloud/stream-applications#160]
For more details, see this: https://github.com/spring-cloud-stream-app-starters/mqtt/issues/18

**Comment by sabbyanandan** [comment]
Probably a similar fix on the tcp-source via https://github.com/spring-cloud/stream-applications/issues/159 applies here also.

**Comment by rjbgaspar** [comment]
Hi  @sobychacko and @sabbyanandan I just checked with version 3.0.1 and the problem still persists,

The client is able to get a connection

**tail -f /mosquitto/log/mosquitto.log**
```
1621841349: New client connected from 172.18.0.1:63296 as scdfClient (p2, c1, k60, u'admin').
```

**Instance mqtt-in.mosquito-inbound logs**
```
    org.springframework.boot.actuate.autoconfigure.web.mappings.MappingsEndpointAutoConfiguration

    org.springframework.boot.autoconfigure.availability.ApplicationAvailabilityAutoConfiguration

    org.springframework.boot.autoconfigure.info.ProjectInfoAutoConfiguration



2021-05-24 07:29:09.624  INFO [mqtt-source,,,] 8054 --- [           main] s.c.s.a.m.s.k.MqttSourceKafkaApplication : Started MqttSourceKafkaApplication in 22.328 seconds (JVM running for 23.836)
```

**Instances for app mqtt-in.log logs**
```
==> standard input <==
```

**Comment by sobychacko** [comment]
@rjbgaspar Sorry about that. We will have a release this week and hopefully, a fix for this will be available in it.

**Comment by leanadvocate** [comment]
> Hi @sobychacko and @sabbyanandan I just checked with version 3.0.1 and the problem still persists,
> 
> The client is able to get a connection
> 
> **tail -f /mosquitto/log/mosquitto.log**
> 
> ```
> 1621841349: New client connected from 172.18.0.1:63296 as scdfClient (p2, c1, k60, u'admin').
> ```
> 
> **Instance mqtt-in.mosquito-inbound logs**
> 
> ```
>     org.springframework.boot.actuate.autoconfigure.web.mappings.MappingsEndpointAutoConfiguration
> 
>     org.springframework.boot.autoconfigure.availability.ApplicationAvailabilityAutoConfiguration
> 
>     org.springframework.boot.autoconfigure.info.ProjectInfoAutoConfiguration
> 
> 
> 
> 2021-05-24 07:29:09.624  INFO [mqtt-source,,,] 8054 --- [           main] s.c.s.a.m.s.k.MqttSourceKafkaApplication : Started MqttSourceKafkaApplication in 22.328 seconds (JVM running for 23.836)
> ```
> 
> **Instances for app mqtt-in.log logs**
> 
> ```
> ==> standard input <==
> ```

is that issue resolved in the new release? Since I too facing the same issue

**Comment by artembilan** [comment]
According to commit tags, the fix has made it into `2020.0.2` release: https://github.com/spring-cloud/stream-applications/commit/2712dab49b5b62b8c63de33c92d68e08fbb6a16b

**Comment by leanadvocate** [comment]
> According to commit tags, the fix has made it into `2020.0.2` release: [2712dab](https://github.com/spring-cloud/stream-applications/commit/2712dab49b5b62b8c63de33c92d68e08fbb6a16b)

Yes it works for me, Thanks
